### PR TITLE
[mp] fix broken links

### PR DESCRIPTION
**Pull request by mattppal**
# Summary
Mintlify pushed a new feature that enables broken link detection via `mintlify broken-links`. It doesn't yet work with CI/CD, so we can't automate this in a precommit hook, but this PR fixes 30 broken links in our docs.

**Before**

image

**After**

image

# Tests
Mintlify CLI

### CODE REVIEW

File: docs/contributing/backend/overview.mdx [hunk 1]
```diff
@@ -60,12 +60,10 @@ root@b9d811e1e3e8:/home/src# python3 -m unittest discover -s mage_ai.tests --fai
 ### API
 
 - [Overview](/api-reference/overview)
-- [Routes](/api-reference/routes)
 
 ### Data integrations
 
 - [Add a new source to data integration pipeline](/data-integrations/sources/add-new-source)
-- [Add a new destination to data integration pipeline](/data-integrations/sources/add-new-destination)
```
Comment (wangxiaoyou1993) [inline docs/contributing/backend/overview.mdx:68]: `/data-integrations/destinations/add-new-destination`

File: docs/contributing/backend/overview.mdx [hunk 2]
```diff
@@ -60,12 +60,10 @@ root@b9d811e1e3e8:/home/src# python3 -m unittest discover -s mage_ai.tests --fai
 ### API
 
 - [Overview](/api-reference/overview)
-- [Routes](/api-reference/routes)
```
Comment (dy46) [inline docs/contributing/backend/overview.mdx:63]: did you mean to delete this?

File: docs/guides/streaming/overview.mdx [hunk 3]
```diff
@@ -18,21 +18,21 @@ pipeline:
 ![Complex streaming pipeline](https://media.graphassets.com/4ZWHyLpgTESu0rB1sXnP)
 
 ## Supported sources
-- [Amazon SQS](sources/amazon-sqs)
```
Comment (wangxiaoyou1993) [inline docs/guides/streaming/overview.mdx:21]: actually, i think the relative paths work. https://docs.mage.ai/guides/streaming/overview#supported-sources

File: docs/production/deploying-to-cloud/azure/setup.mdx [hunk 4]
```diff
@@ -220,7 +220,7 @@ resource "azurerm_container_group" "container_group" {
         You can run the following command to see all the resources that will be created
         by Terraform: `terraform plan`
 
-        By default, here are the **[resources](resources)** that will be
```
Comment (wangxiaoyou1993) [inline docs/production/deploying-to-cloud/azure/setup.mdx:223]: I think this relative path also works

File: docs/contributing/backend/overview.mdx (hunk 2 above)
Comment (mattppal) [reply docs/contributing/backend/overview.mdx:63]: I couldn't find any docs for routes... so I assumed it was a artifact or outdated page

File: docs/production/deploying-to-cloud/azure/setup.mdx (hunk 4 above)
Comment (mattppal) [reply docs/production/deploying-to-cloud/azure/setup.mdx:223]: see comment above

File: docs/production/deploying-to-cloud/azure/setup.mdx [hunk 5]
```diff
@@ -220,7 +220,7 @@ resource "azurerm_container_group" "container_group" {
...
+        By default, here are the **[resources](production/deploying-to-cloud/azure/resources)** that will be
```
Comment (wangxiaoyou1993) [inline docs/production/deploying-to-cloud/azure/setup.mdx:223]: Did you test this url? This looks a relative url. I'm not sure whether it works.

Comment (mattppal) [reply docs/production/deploying-to-cloud/azure/setup.mdx:223]: Nice catch! Fixed

File: docs/production/deploying-to-cloud/gcp/setup.mdx [hunk 6]
```diff
@@ -38,7 +38,7 @@ cd gcp
 terraform plan
 ```
 
-By default, here are the **[resources](/production/deploying-to-cloud/gcp/resources)** that will be
```
Comment (wangxiaoyou1993) [inline docs/production/deploying-to-cloud/gcp/setup.mdx:41]: Why do we update this url? You update it in the wrong doc, I guess.

**Comment by mattppal** [comment]
@wangxiaoyou1993 fixed

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**